### PR TITLE
refactor-only -- keep a Map of (column name -> offsets) around

### PR DESCRIPTION
**Pull request by osheroff**
this PR adds no new functionality, but it speeds up `.findColumnIndex()` which we'll use to generate a PK string.

@vanchi-zendesk

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/Table.java [hunk 1]
```diff
@@ -180,22 +198,7 @@ public String getPKString() {
...
-
-		if ( pkColumnNames == null )
-			return;
-
-		for ( String name : pkColumnNames ) {
-			Integer i = findColumnIndex(name);
-
-			if ( i == -1 ) {
-				throw new RuntimeException("Couldn't find primary key '" + name + "' in table '" + name + "'");
-			}
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/Table.java]: don't know if this check was very important, but it's gone now.

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/java/com/zendesk/maxwell/schema/Table.java [hunk 2]
```diff
@@ -38,15 +42,23 @@ public String getName() {
...
 		for(ColumnDef c : columnList) {
-			if ( c.getName().equals(name) )
-				return i;
-			i++;
+			this.columnOffsetMap.put(c.getName(), i++);
+		}
+	}
+
+	public int findColumnIndex(String name) {
+		initColumnOffsetMap();
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/Table.java]: ok, does this really speed up things? we are doing this again here..

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/Table.java]: oh.  good point, I need to memoize.

**Comment by osheroff** [comment]
memoized.

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**